### PR TITLE
CNN example with Deeplearning4j in Java

### PR DESCRIPTION
**Pull request by HelgaShiryaeva**

### CODE REVIEW

File: deeplearning4j/pom.xml [hunk 1]
```diff
@@ -37,6 +37,16 @@
             <artifactId>deeplearning4j-nn</artifactId>
             <version>${dl4j.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.5</version>
```
Comment (rpvilao) [inline deeplearning4j/pom.xml:43]: create vars for the versions

File: deeplearning4j/src/main/java/com/baeldung/deeplearning4j/cnn/CnnExample.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+import com.baeldung.deeplearning4j.cnn.domain.network.CnnModel;
+import com.baeldung.deeplearning4j.cnn.domain.network.CnnModelProperties;
+import com.baeldung.deeplearning4j.cnn.service.dataset.CifarDataSetService;
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.eval.Evaluation;
+
+@Slf4j
+public class CnnExample {
```
Comment (rpvilao) [inline deeplearning4j/src/main/java/com/baeldung/deeplearning4j/cnn/CnnExample.java:11]: Could we move all the classes to a single package and make them all package protected?

File: deeplearning4j/src/main/java/com/baeldung/deeplearning4j/cnn/domain/network/CnnModel.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                .build();
+
+        network = new MultiLayerNetwork(configuration);
+    }
+
+    public void train() {
+        network.init();
+        int epochsNum = properties.getEpochsNum();
+        IntStream.range(1, epochsNum + 1).forEach(epoch -> {
+            log.info(String.format("Epoch %d?%d", epoch, epochsNum));
```
Comment (rpvilao) [inline deeplearning4j/src/main/java/com/baeldung/deeplearning4j/cnn/domain/network/CnnModel.java:59]: you could use the logging placeholders here ("{}?{}", a, b)

File: deeplearning4j/src/main/java/com/baeldung/deeplearning4j/cnn/domain/network/CnnModel.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import org.nd4j.linalg.lossfunctions.LossFunctions;
+
+import java.util.stream.IntStream;
+
+@Slf4j
+public class CnnModel {
+
+    private final IDataSetService dataSetService;
+
+    private MultiLayerNetwork network;
```
Comment (rpvilao) [inline deeplearning4j/src/main/java/com/baeldung/deeplearning4j/cnn/domain/network/CnnModel.java:24]: this could be final as well right?

**Comment by HelgaShiryaeva** [comment]
Thank you for reviewing!

I made all the required changes, have a look

File: deeplearning4j/pom.xml [hunk 5]
```diff
@@ -37,6 +37,16 @@
             <artifactId>deeplearning4j-nn</artifactId>
             <version>${dl4j.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>${sl4j.version}</version>
```
Comment (rpvilao) [inline deeplearning4j/pom.xml:43]: slf (missing f)

**Pull request merged**